common/predictive-text/worker/model-compositor.ts

```diff
@@ -291,11 +298,21 @@ class ModelCompositor {
 
     // Generate a default 'keep' option if one was not otherwise produced.
     if(!keepOption && keepOptionText != '') {
-      let keepTransform = models.transformToSuggestion(inputTransform, 1);  // 1 is a filler value; goes unused b/c is for a 'keep'.
+      // IMPORTANT:  duplicate the original transform.  Causes nasty side-effects
+      // for context-tracking otherwise!
+      let keepTransform: Transform = {
+        insert: inputTransform.insert,
+        deleteLeft: inputTransform.deleteLeft,
+        deleteRight: inputTransform.deleteRight,
+        id: inputTransform.id
+      };
```

Comment:
Can we just use a spread operator?

```suggestion
      let keepTransform: Transform = { ...inputTransform };
```

That also improves forward compat. Only danger: setters not called, if that's something to be concerned about?